Google.Api.Gax.Grpc/ApiMetadata.cs

```diff
@@ -46,12 +50,21 @@ public sealed partial class ApiMetadata
         /// </summary>
         public bool RequestNumericEnumJsonEncoding { get; }
 
-        private ApiMetadata(string name, Lazy<IReadOnlyList<FileDescriptor>> fileDescriptorsProvider, bool requestNumericEnumJsonEncoding)
+        /// <summary>
+        /// A dictionary (based on ordinal string comparisons) from fully-qualified RPC names
+        /// to byte strings representing overrides for the HTTP rule. This is designed to support
+        /// mixins which are hosted at individual APIs, but which are exposed via different URLs
+        /// to the original mixin definition. This is never null, but may be empty.
+        /// </summary>
+        public IReadOnlyDictionary<string, ByteString> HttpRuleOverrides { get; }
```

Comment:
The problem is, if we did that, then users could mess with it, because protobuf messages are immutable. ByteStrings have the dual benefits of being really easy to generate *and* immutable.